### DISCUSSION TITLE
"signature verification failed" w/ Ed25519, miniflare, and Node >= 16.17

### DISCUSSION THREAD

**Discussion by mhamann** [panva/jose#497] [Q&A]
I'm having an issue verifying JWTs signed with an Ed25519 JWK when using Miniflare. On Node 16.16 everything is fine, but any version of Node greater than that, verification fails.

I think this is closely related to https://github.com/panva/jose/issues/446.

In tracing through the code flow, I can see what the issue is caused by what's coming back from `crypto.importJwk()`.

On Node 16.16, the imported JWK looks approximately like this:
```js
{ name: "NODE-ED25519", namedCurve: "NODE-ED25519" }
```

When that's later passed to `jose.verifyJwt()`, the `namedCurve` property is used to determine how to complete the signature verification.

On Node 16.17 and higher, the imported JWK is missing the `namedCurve` property:
```js
{ name: "NODE-ED25519" }
```

[This PR](https://github.com/nodejs/node/pull/42507) to Node.js seems to be the culprit, but after reading through the changes, I can't isolate exactly what broke this behavior or what can be done to fix it.

Any tips?

**Comment by panva** [comment]
I assume you've tried updating to the latest `miniflare` and the latest lts version of a newer node LTS, e.g. v18, and are using the latest `jose`?

**Reply by mhamann** [reply]
Yes, tested with Node 18.14.0, miniflare v2.11.0, and jose v4.11.2 without success.

**Reply by panva** [reply]
Then I'd suggest to reach out to miniflare first, CI runs fine against `workerd` (effectively CF Workers itself). If there's a problem it's in the runtime.

**Reply by mhamann** [reply]
Will do, thanks!
